qcodes/utils/logger.py

```diff
@@ -0,0 +1,439 @@
+    """
+    Return the provided or default logfile as a pandas DataFrame.
+
+    Unless `qcodes.utils.logger.LOGGING_SEPARATOR` or
+    `qcodes.utils.logger.FORMAT_STRING...` have been changed using the default
+    for the columns and separtor arguments is encouraged.
+
+    Args:
+        logfile: name of the logfile; defaults to current default log file.
+        columns: column headers for the returned dataframe.
+        separator: seperator of the logfile to seperate the columns.
+
+    Retruns:
+        Pandas DataFrame containing the logfile content.
+    """
+    logfile = logfile or get_log_file_name()
+    with open(logfile) as f:
+        raw_cont = f.readlines()
+
+    return log_to_dataframe(raw_cont, columns, separator)
+
+
+def time_difference(firsttimes: Series,
+                    secondtimes: Series,
+                    use_first_series_labels: bool=True) -> Series:
+    """
+    Calculate the time differences between two series
+    containing time stamp strings as their values.
+
+    Args:
+        firsttimes: The oldest times
+        secondtimes: The youngest times
+        use_first_series_labels: If true, the returned Series
+            has the same labels as firsttimes. Else it has
+            the labels of secondtimes
+
+    Returns:
+        A Series with float values of the time difference (s)
+    """
+
+    if ',' in firsttimes.iloc[0]:
+        nfirsttimes = firsttimes.str.replace(',', '.')
+    else:
+        nfirsttimes = firsttimes
+
+    if ',' in secondtimes.iloc[0]:
+        nsecondtimes = secondtimes.str.replace(',', '.')
+    else:
+        nsecondtimes = secondtimes
+
+    t0s = nfirsttimes.astype("datetime64[ns]")
+    t1s = nsecondtimes.astype("datetime64[ns]")
+    timedeltas = (t1s.values - t0s.values).astype('float')*1e-9
+
+    if use_first_series_labels:
+        output = pandas.Series(timedeltas, index=nfirsttimes.index)
+    else:
+        output = pandas.Series(timedeltas, index=nsecondtimes.index)
+
+    return output
+
+
+@contextmanager
+def handler_level(level: LevelType,
+                  handler: Optional[
+                           Union[logging.Handler,
+                                 Sequence[logging.Handler]]]=None):
+    """
+    Context manager to temporarily change the level of handlers.
+    Example:
+        >>> with logger.handler_level(level=logging.DEBUG, handler=[h1, h1]):
+        >>>     root_logger.debug('this is now visible)
+
+    Args:
+        level: level to set the handlers to
+        handler: single or sequence of handlers which to change
+    """
+    if isinstance(handler, logging.Handler):
+        handler = (handler,)
+    original_levels = [h.level for h in handler]
+    for h in handler:
+        h.setLevel(level)
+    yield
+    for h, original_level in zip(handler, original_levels):
+        h.setLevel(original_level)
+
+
+@contextmanager
+def console_level(level: LevelType):
```

Comment:
is there no convenience function for the filelog level so that the filelog always reliable logs everything (..that it is supposed to log)?

Reply:
That was the idea...